Add EDGE, DELAY, DELTA, APPROX_EQUAL switches and TIMER to Logic Conditions

MrD-RC opened the pull request:
This PR adds EDGE, DELAY, DELTA, APPROX EQUAL, and TIMER functionality to logic conditions

### EDGE
EDGE is a momentary logical switch, triggered by `Operand A`. An optional off delay has been added for flexibility. It takes 2 operands:
1. **Operand A**: EDGE Activator. This is a Boolean operand, such as a logical condition or flight mode on/off etc.
2. **Operand B**: _Optional_ On time for EDGE condition [_ms_]. Less than 100ms defaults to instant. If you don't want an off delay, leave `Operand B` as `Value: 0`.

[[Demo]](https://youtu.be/yUoATtfYV2c)

### DELAY
Delay adds a wait before it returns a true value. It takes 2 operands:
1. **Operand A**: DELAY Activator. This is a Boolean operand, such as a logical condition or flight mode on/off etc.
2. **Operand B**: Delay time, before the LC becomes active [_ms_]. The switch in `Operand A` must remain active for the delay to activate after the delay time.

### TIMER
Timer is a simple on/off timer. It works in the same way as OpenTX timers. It takes 2 operands:
1. **Operand A**: Time for the timer to be true [_ms_].
2. **Operand B**: Time for the timer to be false [_ms_].

### APPROX_EQUAL
Approximately equals sees if `Operand B` is within a 1% difference to `Operand A`. It takes 2 operands:
1. **Operand A**: The value that you want to compare.
2. **Operand B**: The value to compare to.

### DELTA
The delta comparator checks to see if `Operand A` has changed by the value in `Operand B` or greater. It takes 2 operands:
1. **Operand A**: The value that you want to compare.
2. **Operand B**: The minimum amount the value must change.

[[DEMO of all these conditions in action]](https://youtu.be/QUepHjfZWSU)

Requires configurator https://github.com/iNavFlight/inav-configurator/pull/1653

MrD-RC committed:
Initial code

MrD-RC committed:
Added EDGE switch to Logical Conditions

DzikuVx commented on docs/Programming Framework.md:
```diff
@@ -80,6 +80,8 @@ IPF can be edited using INAV Configurator user interface, of via CLI
 | 44            | MAX                           | Finds the highest value of `Operand A` and `Operand B` |
 | 45			| FLIGHT_AXIS_ANGLE_OVERRIDE	| Sets the target attitude angle for axis. In other words, when active, it enforces Angle mode (Heading Hold for Yaw) on this axis (Angle mode does not have to be active). `Operand A` defines the axis: `0` - Roll, `1` - Pitch, `2` - Yaw. `Operand B` defines the angle in degrees |
 | 46			| FLIGHT_AXIS_RATE_OVERRIDE	    | Sets the target rate (rotation speed) for axis. `Operand A` defines the axis: `0` - Roll, `1` - Pitch, `2` - Yaw. `Operand B` defines the rate in degrees per second |
+| 47            | EDGE                          | `Operand A` is activation operator [`boolean`], `Operand B` is the time for the edge to stay active [ms]. After activation, operator will return `true` until the time in Operand B is reached |
```
@MrD-RC all good, but technically this is not `EGDE` but `LATCH` or `TIMED_LATCH` or `TEMPORARY_LATCH`. 
`EDGE` would be a detection if `OperandA` value changed more than `OperandB`. 

So maybe let's rename it to a more proper name. And BTW, `EDGE` might be useful too

MrD-RC replied:
Wouldn't that be a delta **|a|>x** switch, if **A** changes more than **B**?

I always took edge to be a momentary switch, activated by operand **A**. I just added an optional off delay to it

Either way, I can add the **A** changes more than **B** comparator too. Happy to name them whatever. 

I actually requested the review to make sure you were happy with how the switch is working, as I would like to add more 😁. x~a and a simple on/off timer are next. I could even add a delay, similar to what is currently named edge, but the time is how long to wait before the switch activates. So long as **A** is true.

MrD-RC committed:
Added some more logic

Added delay, almost equal, and timer. Not yet tested.

MrD-RC committed:
Updates with tested code and docs

Fixed typo
Updated document
Finished & tested LCs for:
- Delay
- Timer
- Delta
- Approx. Equal

MrD-RC committed:
Update Programming Framework.md
